src/pages/LocationPage.js

```diff
@@ -2,27 +2,209 @@
+        var possibleLocations = await maps.geocode({
+            location: {
+                lat: location.coords.latitude,
+                lng: location.coords.longitude,
+            },
+        });
+
+        /* return true if the types includes one of our interesting
+         * component types */
+        function interestingComponent(types: Array<string>): boolean {
+            return !_.isEmpty(_.intersection(
+                types,
+                ['locality', 'administrative_area_level_1']
+            ));
+        }
+
+        for (var geocodedLocation of possibleLocations) {
+            if (_.contains(geocodedLocation.types, 'locality')) {
+                /* build a location name from the address components specified
+                 * in interestingComponent. We do this because we don't want
+                 * to show all the parts of Google's formatted_address */
+                var name = [
+                    /*::`*/
+                    for (component of geocodedLocation.address_components)
+                    if (interestingComponent(component.types))
+                        component.short_name
+                    /*::`*/
+                ].join(' ');
+
+                return {
+                    location: location,
+                    name: name,
+                };
+            }
+        }
+
+        throw "Unable to determine your suburb";
+    }
+
+    onGeolocationTouchTap(): void {
+        if (this.state.geolocation != GeoLocationState.NOT_STARTED) {
+            return;
+        }
+
+        this.setState({
+            geolocation: GeoLocationState.RUNNING,
+        });
+
+        this.locateMe()
+            .then(params => {
+                var { location, name } = params;
+
+                this.setState({
+                    geolocation: GeoLocationState.COMPLETE,
+                    locationName: name,
+                    locationCoords: location,
+                });
+            })
+
+            .catch((e) => {
+                console.error(e);
+                this.setState({
+                    geolocation: GeoLocationState.FAILED,
+                    error: e.message,
+                });
+            });
+    }
+
+    onTouchDoneButton(event: Event): void {
+        sessionstorage.setItem('location', this.state.locationName);
+        this.replaceWith(this.getQuery().next);
+    }
+
+    onSearchChange(event: Event): void {
+        if (event.target instanceof HTMLInputElement) {
+            this.setState({
+                locationName: event.target.value,
+            });
+        }
+    }
+
+    componentDidMount(): void {
+        this.setState({
+            locationName: sessionstorage.getItem('location'),
+        });
     }
 
     render(): React.Element {
         return (
-            <div>
-                <mui.AppBar title="Personalise" />
+            <div className="LocationPage">
+                <mui.AppBar
+                    title="Personalise"
```

Comment:
Again - not sure if this is the HTML attr "title" - but would a screen reader pick this up?